src/components/search/search-bar.tsx

```diff
@@ -148,37 +148,36 @@ export const SearchBar: FunctionComponent<SearchBarProps> = ({ inputRef }) => {
     );
 
     const handleMatchingElement = useCallback(
-        (event: SyntheticEvent, data: matchingElementProps | string | null) => {
+        async (
+            event: SyntheticEvent,
+            data: matchingElementProps | string | null
+        ) => {
             const matchingElement = elementsFound.find(
                 (element: matchingElementProps) => element === data
             );
             if (matchingElement !== undefined) {
                 const elementUuidPath = matchingElement?.pathUuid;
-                const promises = elementUuidPath.map((e: string) => {
-                    return fetchDirectoryContent(e as UUID)
-                        .then((res) => {
-                            updateMapData(
-                                e,
-                                res.filter(
-                                    (res) => res.type === ElementType.DIRECTORY
-                                ) as IDirectory[]
-                            );
-                        })
-                        .catch((error) =>
-                            snackError({
-                                messageTxt: error.message,
-                                headerId: 'pathRetrievingError',
-                            })
+                try {
+                    for (const uuid of elementUuidPath) {
```

Comment:
If we don't fetch the last one, then `handleDispatchDirectory` and following code will check first if selectedDirectory is in the map. If not, it will avoid select current directory.
By the way, selecting the current directory proc another fetchDirectoryContent then we have 0 or two identical fetch.
It was already like that, will refactor later.